### ISSUE TITLE
Amaze crashes when 'Preselected config' is pressed

### ISSUE THREAD

**Issue by Raph-2018** [TeamAmaze/AmazeFileManager#1395]
Hello,
I noticed a bug with Amaze 3.3.0 RC1 on Android 4.2.2.
When I press 'Preselected config' in the 'Select color config' menu, Amaze crashes. This is a new bug, as it was working perfectly with Amaze 3.2.2.
Thanks for your work on this very useful app :)

**Comment by EmmanuelMess** [comment]
I will need the logs for the crash.

**Comment by Raph-2018** [comment]
@VishalNehra I have no 'report' button in the crash dialog, only a 'ok' button with no logs displayed... Please let me know if there is anything I can do to get the logs!
Note that I downloaded version 3.3.0 RC1 from an apk found on the web, as the version available on the Play Store is still 3.2.2. I do not know were to find a more 'official' apk of such a release candidate (I found nothing on github). Thus, I hope the file I used was not corrupted (md5 is: 5e80e14b314c461ddd1bf9d1a5a18b1b). However, as the file was available on multiple websites, I have installed from three different files, and I always have the same bug.
Maybe some more informations on my device can be helpful: I am using a Acer Liquid Z130 (not rooted), with Android 4.2.2.

**Comment by EmmanuelMess** [comment]
@Raph-2018 I think there is a beta program in Play Store

**Comment by Raph-2018** [comment]
Unfortunately, I was unable to install the beta version from the Play Store. Google is complaining that I have not enough available space on my phone, although I have more than 20Go on my external sd card and about 400Mo on my internal memory! Furthermore, Google tells me that I would need 108 Mo to install Amaze beta, but the apk I dowloaded is only 7.4 Mo ?!
Maybe you could try to reproduce the bug on your side with a virtual machine of something (I do not know if it is possible, as I am not a developer)...

**Comment by EmmanuelMess** [comment]
> Maybe you could try to reproduce the bug on your side with a virtual machine of something (I do not know if it is possible, as I am not a developer)...

That's part of the testing done before adding the code, but no crashes seem to have pop up...

**Comment by VishalNehra** [comment]
@Raph-2018 you can also use logging apps like alogcat and get us the log.

**Comment by Raph-2018** [comment]
I have managed to get some logs with MatLog. During the crash, I got:

08-16 22:37:36.120 E/dalvikvm(10175): Could not find class 'android.app.ActivityManager$TaskDescription', referenced from method com.amaze.filemanager.activities.MainActivity.onCreate
08-16 22:37:36.557 E/dalvikvm(10175): Could not find class 'android.support.v4.widget.DrawerLayout$1', referenced from method android.support.v4.widget.DrawerLayout.<init>
08-16 22:37:36.563 E/dalvikvm(10175): Could not find class 'android.view.WindowInsets', referenced from method android.support.v4.widget.DrawerLayout.onDraw
08-16 22:37:36.566 E/dalvikvm(10175): Could not find class 'android.view.WindowInsets', referenced from method android.support.v4.widget.DrawerLayout.onMeasure
08-16 22:37:36.566 E/dalvikvm(10175): Could not find class 'android.view.WindowInsets', referenced from method android.support.v4.widget.DrawerLayout.onMeasure
08-16 22:37:36.642 E/dalvikvm(10175): Could not find class 'android.graphics.drawable.RippleDrawable', referenced from method android.support.v7.widget.AppCompatImageHelper.hasOverlappingRendering
08-16 22:37:37.548 E/RecyclerView(10175): No adapter attached; skipping layout
08-16 22:37:37.550 E/RecyclerView(10175): No adapter attached; skipping layout
08-16 22:37:41.195 E/dalvikvm(10175): Could not find class 'android.app.ActivityManager$TaskDescription', referenced from method com.amaze.filemanager.activities.PreferencesActivity.invalidateRecentsColorAndIcon
08-16 22:37:41.230 E/dalvikvm(10175): Could not find class 'android.hardware.fingerprint.FingerprintManager', referenced from method com.amaze.filemanager.fragments.preference_fragments.PrefFrag.onCreate
08-16 22:37:48.948 E/AndroidRuntime(10175): FATAL EXCEPTION: main
08-16 22:37:48.948 E/AndroidRuntime(10175): android.view.InflateException: Binary XML file line #2: Error inflating class <unknown>
08-16 22:37:48.948 E/AndroidRuntime(10175): 	at android.view.LayoutInflater.createView(LayoutInflater.java:613)
08-16 22:37:48.948 E/AndroidRuntime(10175): 	at com.android.internal.policy.impl.PhoneLayoutInflater.onCreateView(PhoneLayoutInflater.java:56)
08-16 22:37:48.948 E/AndroidRuntime(10175): 	at android.view.LayoutInflater.onCreateView(LayoutInflater.java:660)
08-16 22:37:48.948 E/AndroidRuntime(10175): 	at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:685)
08-16 22:37:48.948 E/AndroidRuntime(10175): 	at android.view.LayoutInflater.inflate(LayoutInflater.java:466)
08-16 22:37:48.948 E/AndroidRuntime(10175): 	at android.view.LayoutInflater.inflate(LayoutInflater.java:396)
08-16 22:37:48.948 E/AndroidRuntime(10175): 	at android.view.LayoutInflater.inflate(LayoutInflater.java:352)
08-16 22:37:48.948 E/AndroidRuntime(10175): 	at android.preference.DialogPreference.onCreateDialogView(DialogPreference.java:343)
08-16 22:37:48.948 E/AndroidRuntime(10175): 	at android.preference.DialogPreference.showDialog(DialogPreference.java:287)
08-16 22:37:48.948 E/AndroidRuntime(10175): 	at com.amaze.filemanager.ui.dialogs.ColorPickerDialog.showDialog(ColorPickerDialog.java:198)
08-16 22:37:48.948 E/AndroidRuntime(10175): 	at android.preference.DialogPreference.onClick(DialogPreference.java:266)
08-16 22:37:48.948 E/AndroidRuntime(10175): 	at android.preference.Preference.performClick(Preference.java:949)
08-16 22:37:48.948 E/AndroidRuntime(10175): 	at android.preference.PreferenceScreen.onItemClick(PreferenceScreen.java:215)
08-16 22:37:48.948 E/AndroidRuntime(10175): 	at android.widget.AdapterView.performItemClick(AdapterView.java:298)
08-16 22:37:48.948 E/AndroidRuntime(10175): 	at android.widget.AbsListView.performItemClick(AbsListView.java:1128)
08-16 22:37:48.948 E/AndroidRuntime(10175): 	at android.widget.AbsListView$PerformClick.run(AbsListView.java:2815)
08-16 22:37:48.948 E/AndroidRuntime(10175): 	at android.widget.AbsListView$1.run(AbsListView.java:3574)
08-16 22:37:48.948 E/AndroidRuntime(10175): 	at android.os.Handler.handleCallback(Handler.java:800)
08-16 22:37:48.948 E/AndroidRuntime(10175): 	at android.os.Handler.dispatchMessage(Handler.java:100)
08-16 22:37:48.948 E/AndroidRuntime(10175): 	at android.os.Looper.loop(Looper.java:194)
08-16 22:37:48.948 E/AndroidRuntime(10175): 	at android.app.ActivityThread.main(ActivityThread.java:5371)
08-16 22:37:48.948 E/AndroidRuntime(10175): 	at java.lang.reflect.Method.invokeNative(Native Method)
08-16 22:37:48.948 E/AndroidRuntime(10175): 	at java.lang.reflect.Method.invoke(Method.java:525)
08-16 22:37:48.948 E/AndroidRuntime(10175): 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:833)
08-16 22:37:48.948 E/AndroidRuntime(10175): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:600)
08-16 22:37:48.948 E/AndroidRuntime(10175): 	at dalvik.system.NativeStart.main(Native Method)
08-16 22:37:48.948 E/AndroidRuntime(10175): Caused by: java.lang.reflect.InvocationTargetException
08-16 22:37:48.948 E/AndroidRuntime(10175): 	at java.lang.reflect.Constructor.constructNative(Native Method)
08-16 22:37:48.948 E/AndroidRuntime(10175): 	at java.lang.reflect.Constructor.newInstance(Constructor.java:417)
08-16 22:37:48.948 E/AndroidRuntime(10175): 	at android.view.LayoutInflater.createView(LayoutInflater.java:587)
08-16 22:37:48.948 E/AndroidRuntime(10175): 	... 25 more
08-16 22:37:48.948 E/AndroidRuntime(10175): Caused by: java.lang.UnsupportedOperationException: Can't convert to dimension: type=0x2
08-16 22:37:48.948 E/AndroidRuntime(10175): 	at android.content.res.TypedArray.getDimensionPixelSize(TypedArray.java:463)
08-16 22:37:48.948 E/AndroidRuntime(10175): 	at android.view.View.<init>(View.java:3364)
08-16 22:37:48.948 E/AndroidRuntime(10175): 	at android.view.ViewGroup.<init>(ViewGroup.java:436)
08-16 22:37:48.948 E/AndroidRuntime(10175): 	at android.widget.LinearLayout.<init>(LinearLayout.java:176)
08-16 22:37:48.948 E/AndroidRuntime(10175): 	at android.widget.LinearLayout.<init>(LinearLayout.java:172)
08-16 22:37:48.948 E/AndroidRuntime(10175): 	... 28 more
08-16 22:37:48.981 E/AppErrorDialog(510): Failed to get ILowStorageHandle instance

I hope that it can be of some help to fix the bug! 
One more thing: all elements in the 'Select color config' menu are disabled (grayed), except 'Preselected config'. I am unsure if that is the expected behavior, as I cannot access the 'Preselected config' menu (because of the crash). Is there some 'custom colors' option in that sub-menu to activate the elements in the 'Select color config' menu?

**Comment by EmmanuelMess** [comment]
> Is there some 'custom colors' option in that sub-menu to activate the elements in the 'Select color config' menu?

Yes.

**Comment by Raph-2018** [comment]
@EmmanuelMess Perfect, so the disabled menu is expected by default.
Another interesting observation: if I rotate the screen when I am in the 'Select color config' menu, Amaze also crashes. It does not happen in the other menus of the app, so it is probably a side-effect of the same bug. During this new crash, I got the following logs:

08-17 14:21:25.273 E/AndroidRuntime(4740): FATAL EXCEPTION: main
08-17 14:21:25.273 E/AndroidRuntime(4740): java.lang.RuntimeException: Unable to start activity ComponentInfo{com.amaze.filemanager/com.amaze.filemanager.activities.PreferencesActivity}: java.lang.NullPointerException
08-17 14:21:25.273 E/AndroidRuntime(4740): 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2343)
08-17 14:21:25.273 E/AndroidRuntime(4740): 	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2395)
08-17 14:21:25.273 E/AndroidRuntime(4740): 	at android.app.ActivityThread.handleRelaunchActivity(ActivityThread.java:3902)
08-17 14:21:25.273 E/AndroidRuntime(4740): 	at android.app.ActivityThread.access$700(ActivityThread.java:162)
08-17 14:21:25.273 E/AndroidRuntime(4740): 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1370)
08-17 14:21:25.273 E/AndroidRuntime(4740): 	at android.os.Handler.dispatchMessage(Handler.java:107)
08-17 14:21:25.273 E/AndroidRuntime(4740): 	at android.os.Looper.loop(Looper.java:194)
08-17 14:21:25.273 E/AndroidRuntime(4740): 	at android.app.ActivityThread.main(ActivityThread.java:5371)
08-17 14:21:25.273 E/AndroidRuntime(4740): 	at java.lang.reflect.Method.invokeNative(Native Method)
08-17 14:21:25.273 E/AndroidRuntime(4740): 	at java.lang.reflect.Method.invoke(Method.java:525)
08-17 14:21:25.273 E/AndroidRuntime(4740): 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:833)
08-17 14:21:25.273 E/AndroidRuntime(4740): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:600)
08-17 14:21:25.273 E/AndroidRuntime(4740): 	at dalvik.system.NativeStart.main(Native Method)
08-17 14:21:25.273 E/AndroidRuntime(4740): Caused by: java.lang.NullPointerException
08-17 14:21:25.273 E/AndroidRuntime(4740): 	at com.amaze.filemanager.ui.dialogs.ColorPickerDialog.select(ColorPickerDialog.java:164)
08-17 14:21:25.273 E/AndroidRuntime(4740): 	at com.amaze.filemanager.ui.dialogs.ColorPickerDialog.onRestoreInstanceState(ColorPickerDialog.java:272)
08-17 14:21:25.273 E/AndroidRuntime(4740): 	at android.preference.Preference.dispatchRestoreInstanceState(Preference.java:1757)
08-17 14:21:25.273 E/AndroidRuntime(4740): 	at android.preference.PreferenceGroup.dispatchRestoreInstanceState(PreferenceGroup.java:327)
08-17 14:21:25.273 E/AndroidRuntime(4740): 	at android.preference.Preference.restoreHierarchyState(Preference.java:1739)
08-17 14:21:25.273 E/AndroidRuntime(4740): 	at android.preference.PreferenceFragment.onActivityCreated(PreferenceFragment.java:181)
08-17 14:21:25.273 E/AndroidRuntime(4740): 	at android.app.Fragment.performActivityCreated(Fragment.java:1703)
08-17 14:21:25.273 E/AndroidRuntime(4740): 	at android.app.FragmentManagerImpl.moveToState(FragmentManager.java:903)
08-17 14:21:25.273 E/AndroidRuntime(4740): 	at android.app.FragmentManagerImpl.moveToState(FragmentManager.java:1057)
08-17 14:21:25.273 E/AndroidRuntime(4740): 	at android.app.FragmentManagerImpl.moveToState(FragmentManager.java:1039)
08-17 14:21:25.273 E/AndroidRuntime(4740): 	at android.app.FragmentManagerImpl.dispatchActivityCreated(FragmentManager.java:1840)
08-17 14:21:25.273 E/AndroidRuntime(4740): 	at android.app.Activity.performCreate(Activity.java:5125)
08-17 14:21:25.273 E/AndroidRuntime(4740): 	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1081)
08-17 14:21:25.273 E/AndroidRuntime(4740): 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2307)
08-17 14:21:25.273 E/AndroidRuntime(4740): 	... 12 more
08-17 14:21:25.304 E/AppErrorDialog(510): Failed to get ILowStorageHandle instance

By the way, I also noticed that both 'Select color config' and 'Preselected configs' menu entries are not yet translated into French (my language). Here is my translation for these items:
"Select color config" → "Configurer les couleurs"
"Preselected configs" → "Couleurs préconfigurées"
